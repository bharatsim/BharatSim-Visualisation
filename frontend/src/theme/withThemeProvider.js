import React from 'react';
import { MuiThemeProvider, CssBaseline } from '@material-ui/core';

import theme from './theme';

function withThemeProvider(WrappedComponent) {
  return function WithTheme(props) {
    return (
      <MuiThemeProvider theme={theme}>
        <CssBaseline />
        <WrappedComponent {...props} />
      </MuiThemeProvider>
    );
  };
}

export default withThemeProvider;
