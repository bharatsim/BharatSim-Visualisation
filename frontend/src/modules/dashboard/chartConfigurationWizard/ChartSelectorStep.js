import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { Box, Button, Divider, Typography } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';

import ImageOption from '../../../uiComponent/ImageOption';
import theme from '../../../theme/theme';
import chartConfigs from '../../../config/chartConfigs';
import { useFooterStyles } from './styles';

const useStyles = makeStyles(() => ({
  chartSelectorContainer: {
    margin: theme.spacing(6, 0),
  },
}));

function ChartSelectorStep({ onNext, chart }) {
  const classes = useStyles();
  const footerClasses = useFooterStyles();
  const [selectedChart, setSelectedChart] = useState(chart);
  function onChartClick(selectedChartValue) {
    setSelectedChart(selectedChartValue);
  }

  function onNextCLick() {
    onNext(selectedChart);
  }

  const isNextDisable = !selectedChart;

  return (
    <Box mt={2}>
      <Divider />
      <Box px={2} py={4}>
        <Typography variant="subtitle1">
          Choose a chart type to start configuring your chart
        </Typography>
        <Box className={classes.chartSelectorContainer}>
          {Object.values(chartConfigs).map((chartConfig) => {
            return (
              <Box key={chartConfig.label}>
                <Box mb={2}>
                  <Typography variant="subtitle2">{chartConfig.label}</Typography>
                </Box>
                <Box>
                  <ImageOption
                    dataTestId={chartConfig.key}
                    value={chartConfig.key}
                    label={chartConfig.label}
                    icon={<img src={chartConfig.icon} alt={chartConfig.key} />}
                    isSelected={selectedChart === chartConfig.key}
                    onCLick={onChartClick}
                  />
                </Box>
              </Box>
            );
          })}
        </Box>
      </Box>
      <Box className={footerClasses.footer}>
        <Divider />
        <Box mt={3} display="flex" justifyContent="flex-end">
          <Button
            variant="contained"
            color="primary"
            size="small"
            onClick={onNextCLick}
            disabled={isNextDisable}
          >
            Next
          </Button>
        </Box>
      </Box>
    </Box>
  );
}

ChartSelectorStep.defaultProps = {
  chart: '',
};

ChartSelectorStep.propTypes = {
  onNext: PropTypes.func.isRequired,
  chart: PropTypes.string,
};

export default ChartSelectorStep;
