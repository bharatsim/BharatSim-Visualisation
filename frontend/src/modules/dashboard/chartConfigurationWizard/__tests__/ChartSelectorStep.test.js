import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import ChartSelectorStep from '../ChartSelectorStep';
import withThemeProvider from '../../../../theme/withThemeProvider';

const ComponentWithProvider = withThemeProvider(({ onNext, chart }) => (
  <ChartSelectorStep onNext={onNext} chart={chart} />
));

describe('Chart configuration wizard', () => {
  it('should match snapshot', () => {
    const { container } = render(<ComponentWithProvider onNext={jest.fn()} chart="" />);

    expect(container).toMatchSnapshot();
  });
  it('should select chart and pass selected chart to on next callback', () => {
    const onNext = jest.fn();
    const { getByText } = render(<ComponentWithProvider onNext={onNext} chart="" />);

    const lineChartOption = getByText('Line Chart');
    const nextButton = getByText('Next');

    fireEvent.click(lineChartOption);
    fireEvent.click(nextButton);

    expect(onNext).toHaveBeenCalledWith('lineChart');
  });

  it('should disable next button if not chart is selected', () => {
    const onNext = jest.fn();
    const { getByText } = render(<ComponentWithProvider onNext={onNext} chart="" />);

    const nextButton = getByText('Next').closest('button');

    expect(nextButton).toBeDisabled();
  });

  it('should call next with initial selected chart', () => {
    const onNext = jest.fn();
    const { getByText } = render(<ComponentWithProvider onNext={onNext} chart="Bar Chart" />);

    const nextButton = getByText('Next');

    fireEvent.click(nextButton);

    expect(onNext).toHaveBeenCalledWith('Bar Chart');
  });
});
