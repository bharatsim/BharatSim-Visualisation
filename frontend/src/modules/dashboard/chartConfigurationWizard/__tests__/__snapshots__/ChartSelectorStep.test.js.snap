// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chart configuration wizard should match snapshot 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-3"
  >
    <hr
      class="MuiDivider-root"
    />
    <div
      class="MuiBox-root MuiBox-root-4"
    >
      <h6
        class="MuiTypography-root MuiTypography-subtitle1"
      >
        Choose a chart type to start configuring your chart
      </h6>
      <div
        class="MuiBox-root MuiBox-root-5 makeStyles-chartSelectorContainer-1"
      >
        <div
          class="MuiBox-root MuiBox-root-6"
        >
          <div
            class="MuiBox-root MuiBox-root-7"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle2"
            >
              Line Chart
            </h6>
          </div>
          <div
            class="MuiBox-root MuiBox-root-8"
          >
            <div
              class="MuiBox-root MuiBox-root-13 makeStyles-buttonContainer-9 makeStyles-buttonContainer-12"
              data-testid="lineChart"
            >
              <div
                class="MuiBox-root MuiBox-root-14 makeStyles-imageContainer-10 null"
              >
                <img
                  alt="lineChart"
                  src="lineChart.svg"
                />
              </div>
              <span
                class="MuiTypography-root MuiTypography-caption"
              >
                Line Chart
              </span>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root MuiBox-root-15"
        >
          <div
            class="MuiBox-root MuiBox-root-16"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle2"
            >
              Bar Chart
            </h6>
          </div>
          <div
            class="MuiBox-root MuiBox-root-17"
          >
            <div
              class="MuiBox-root MuiBox-root-19 makeStyles-buttonContainer-9 makeStyles-buttonContainer-18"
              data-testid="barChart"
            >
              <div
                class="MuiBox-root MuiBox-root-20 makeStyles-imageContainer-10 null"
              >
                <img
                  alt="barChart"
                  src="barChart.svg"
                />
              </div>
              <span
                class="MuiTypography-root MuiTypography-caption"
              >
                Bar Chart
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiBox-root MuiBox-root-21 makeStyles-footer-2"
    >
      <hr
        class="MuiDivider-root"
      />
      <div
        class="MuiBox-root MuiBox-root-22"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall Mui-disabled Mui-disabled"
          disabled=""
          tabindex="-1"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Next
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
